### DISCUSSION TITLE
Handle external link click

### DISCUSSION THREAD

**Discussion by vitaliemiron** [vercel/next.js#17108] [Help]
**# Feature request**

- Is there a  method to handle click on external url?

**## Is your feature request related to a problem? Please describe.**

- When user clicks an external url on the whole site I need to show them a popup where user can confirm/cancel site leaving.

**A clear and concise description of what you want and what your use case is.**

- All links on  my site are wrapped by <Link> component from nextjs. But I also have a html editor on backend side where client can add their own links in the text. So that sections on site aren't wrapped by Link component.

**## Describe the solution you'd like**

- I  want to catch site leaving and possibility to continue/cancel action.

**## Describe alternatives you've considered**

- Is possible to create a component for links and wrap all the links in that, but I didn't consider this as a good solution.

**Comment by M1ck0** [comment]
The only thing that crosses my mind is what you can do, is to check what the user has clicked on and then call `onBeforeUnload` to show him some kind of message

Read more [here](https://www.npmjs.com/package/react-beforeunload) about package.

I am not sure if there are any easy ways to do this since browsers do not give you the info of where user is navigating to, or if he just refreshed the page or smth else.

**Reply by vitaliemiron** [reply]
Thank you @M1ck0, I think in most cases your solution will work but I need to show a custom popup windows. ```onBeforeUnload``` din't support custom messages, just browser default behavior.
